### PR TITLE
Upgrade slinky to 1.0.8

### PR DESCRIPTION
**Pull request by chenyaoy**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Enhanced CI/CD automation by adding triggers for the `chenyao/upgrade-slinky-to-1.0.8` branch in multiple workflows.
  
- **Improvements**
	- Updated the Dockerfile to use the latest Golang Alpine image, ensuring better performance and security.
	- Simplified the `migratePricesToMarketMap` function to improve efficiency by directly invoking methods without unnecessary handler initialization.
	- Streamlined the `ConstructMarketMapFromParams` function for better performance by removing the API handler instantiation.

- **Dependency Updates**
	- Upgraded various dependencies in the `go.mod` file to their latest versions, ensuring compatibility and access to new features and improvements.

### CODE REVIEW

File: protocol/go.mod [hunk 1]
```diff
@@ -1,19 +1,17 @@
 module github.com/dydxprotocol/v4-chain/protocol
 
-go 1.22
-
-toolchain go1.22.0
+go 1.22.2
```
Comment (vincentwschau) [inline protocol/go.mod:3]: why does this get removed?

File: protocol/go.mod [hunk 2]
```diff
@@ -1,19 +1,17 @@
 module github.com/dydxprotocol/v4-chain/protocol
 
-go 1.22
-
-toolchain go1.22.0
```
Comment (vincentwschau) [inline protocol/go.mod:5]: Why does this get removed?

File: .github/workflows/protocol-build-and-push-snapshot.yml [hunk 3]
```diff
@@ -6,6 +6,7 @@ on:  # yamllint disable-line rule:truthy
       - main
       - 'release/protocol/v[0-9]+.[0-9]+.x'  # e.g. release/protocol/v0.1.x
       - 'release/protocol/v[0-9]+.x'  # e.g. release/protocol/v1.x
+      - 'chenyao/upgrade-slinky-to-1.0.8'
```
Comment (vincentwschau) [inline .github/workflows/protocol-build-and-push-snapshot.yml:9]: nit:remove this testing config?

File: .github/workflows/indexer-build-and-push-dev-staging.yml [hunk 4]
```diff
@@ -6,6 +6,7 @@ on: # yamllint disable-line rule:truthy
       - main
       - 'release/indexer/v[0-9]+.[0-9]+.x'  # e.g. release/indexer/v0.1.x
       - 'release/indexer/v[0-9]+.x'  # e.g. release/indexer/v1.x
+      - 'chenyao/upgrade-slinky-to-1.0.8'
```
Comment (vincentwschau) [inline .github/workflows/indexer-build-and-push-dev-staging.yml:9]: nit:remove this testing config?

**Review by vincentwschau — APPROVED** [review]

File: protocol/go.mod (hunk 2 above)
Comment (chenyaoy) [reply protocol/go.mod:5]: I set slinky version to v1.0.8 and ran go mod tidy and it made this change

**Review by Eric-Warehime — APPROVED** [review]

**Comment by chenyaoy** [comment]
@mergifyio backport release/protocol/v6.x

**Pull request merged**